### PR TITLE
Adding forces to SPME

### PR DESCRIPTION
**Pull request by sfogerty**
This adds force calculation to the SPME example, where before only energies were computed.

### CODE REVIEW

File: core/example/longrange/spme.cpp [hunk 1]
```diff
@@ -764,6 +812,122 @@ double TPME::compute( ParticleList &particles, ParticleList &mesh, double lx,
...
+                 std::abs( meshr( idx, 2 ) - ( r( pidx, 2 ) + 1.0 ) ) )
+            {
+                closestpart = r( pidx, 2 ) + 1.0;
+            }
+            if ( std::abs( zdist ) >
+                 std::abs( meshr( idx, 2 ) - ( r( pidx, 2 ) - 1.0 ) ) )
+            {
+                closestpart = r( pidx, 2 ) - 1.0;
+            }
+            zdist = closestpart - meshr( idx, 2 );
```
Comment (dalg24) [inline core/example/longrange/spme.cpp]: This introduces a lot of code duplication.

File: core/example/longrange/spme.cpp [hunk 2]
```diff
@@ -764,6 +812,122 @@ double TPME::compute( ParticleList &particles, ParticleList &mesh, double lx,
...
+                    q( pidx ) *
+                    TPME::oneDspline( 2.0 - ( std::abs( xdist ) / spacing ) ) *
+                    TPME::oneDsplinederiv( ydist / spacing ) *
+                    TPME::oneDspline( 2.0 - ( std::abs( zdist ) / spacing ) ) *
+                    Qktest[idx][0];
+                f( pidx, 2 ) +=
+                    q( pidx ) *
+                    TPME::oneDspline( 2.0 - ( std::abs( xdist ) / spacing ) ) *
+                    TPME::oneDspline( 2.0 - ( std::abs( ydist ) / spacing ) ) *
+                    TPME::oneDsplinederiv( zdist / spacing ) * Qktest[idx][0];
```
Comment (dalg24) [inline core/example/longrange/spme.cpp]: Please format these three statement the same way.

File: core/example/longrange/spme.cpp [hunk 3]
```diff
@@ -764,6 +812,122 @@ double TPME::compute( ParticleList &particles, ParticleList &mesh, double lx,
...
+                    std::cout
+                        << q( pidx ) *
+                               TPME::oneDspline(
+                                   2.0 - ( std::abs( xdist ) / spacing ) ) *
+                               TPME::oneDsplinederiv( ydist / spacing ) *
+                               TPME::oneDspline(
+                                   2.0 - ( std::abs( zdist ) / spacing ) ) *
+                               Qktest[idx][0]
+                        << std::endl;
+                }
```
Comment (dalg24) [inline core/example/longrange/spme.cpp]: This won't work with CUDA.  Did you mean to commit that debug print statement?

**Comment by sslattery** [comment]
Closing in lieu of solver work in CabanaMD

**Pull request closed without merging**